### DISCUSSION TITLE
Feature Request: Option to Limit Output to Code Snippets Instead of Editing Whole File

### DISCUSSION THREAD

**Discussion by kcui23** [cline/cline#579] [Feature Requests]
**Description**:

Hi team,
First of all, thank you for developing such a powerful tool like cline! I’ve been using it to modify and generate code using LLM APIs, and overall, it’s been quite helpful.

However, I’ve encountered a recurring issue when working with large code files. It’s a well-known problem that the tool sometimes truncates outputs in large files, and to be honest, I’m not surprised by that, nor do I find it particularly bothersome.

What I do find challenging is when I specifically know I need to modify just a small portion—say, a few dozen lines of a large file—but the `edit file` tool still gets invoked to update the entire file from start to finish, even if 99% of the code will remain unchanged. This ends up consuming a lot of time.

**Request**:

Is there a way to force cline to avoid using the `edit file` tool and instead generate code snippets directly in the chat, which I could then copy and paste into the actual file myself? Having this option would greatly improve my workflow by saving time when only small modifications are needed in large codebases.

Looking forward to your response!

Thank you!

**Comment by PhunkyBob** [comment]
Would my suggestion answer your request? 
https://github.com/cline/cline/discussions/473#discussioncomment-11009830

It seems like you, as a developer I would prefer say "do this right here" instead of "I want this, help yourself".

**Reply by kcui23** [reply]
exactly, i want the same thing

**Comment by kcui23** [comment]
Better than nothing: I just discovered that cline *seriously* refuses to edit my file if it's told it'll get fired for doing so. 😂 Found that out by accident—lol!
